format/FlightSql.proto

```diff
@@ -867,6 +867,151 @@ enum SqlSupportsConvert {
   SQL_CONVERT_VARCHAR = 19;
 }
 
+enum DATE_SUBCODE {
+  UNKNOWN_SUBCODE = 0;
+  SQL_CODE_DATE = 1;
+  SQL_CODE_TIME = 2;
+  SQL_CODE_TIMESTAMP = 3;
+}
+
+enum SQL_DATA_TYPE {
+  UNKNOWN = 0;
+  SQL_CODE_YEAR = 1;
+  SQL_CODE_MONTH = 2;
+  SQL_CODE_DAY = 3;
+  SQL_CODE_HOUR = 4;
+  SQL_CODE_MINUTE = 5;
+  SQL_CODE_SECOND = 6;
+  SQL_CODE_YEAR_TO_MONTH = 7;
+  SQL_CODE_DAY_TO_HOUR = 8;
+  SQL_CODE_DAY_TO_MINUTE = 9;
+  SQL_CODE_DAY_TO_SECOND = 10;
+  SQL_CODE_HOUR_TO_MINUTE = 11;
+  SQL_CODE_HOUR_TO_SECOND = 12;
+  SQL_CODE_MINUTE_TO_SECOND = 13;
+  SQL_INTERVAL_YEAR = 101;
+  SQL_INTERVAL_MONTH = 102;
+  SQL_INTERVAL_DAY = 103;
+  SQL_INTERVAL_HOUR = 104;
+  SQL_INTERVAL_MINUTE = 105;
+  SQL_INTERVAL_SECOND = 106;
+  SQL_INTERVAL_YEAR_TO_MONTH = 107;
+  SQL_INTERVAL_DAY_TO_HOUR = 108;
+  SQL_INTERVAL_DAY_TO_MINUTE = 109;
+  SQL_INTERVAL_DAY_TO_SECOND = 110;
+  SQL_INTERVAL_HOUR_TO_MINUTE = 111;
+  SQL_INTERVAL_HOUR_TO_SECOND = 112;
+  SQL_INTERVAL_MINUTE_TO_SECOND = 113;
+}
+
+enum Nullable {
+  /**
+   * Indicates that the fields does not allow the use of null values.
+   */
+  TYPE_NO_NULLS = 0;
```

Comment:
Similarly, while ODBC/JDBC might have certain names, we don't have to stick to those names 1:1 right? This could be NULLABILITY_NO_NULLS, NULLABILITY_NULLABLE, NULLABILITY_UNKNOWN (and usually UNKNOWN should be the zero value, unless the exact value is important here)